src/main/java/by/exadel/internship/service/impl/UserTimeSlotServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Restoring time with uuid = {} for user with uuid ={}", time.getId(), time.getUserId());
+        UserDTO userDTO = userService.getById(time.getUserId());
+        UserTimeSlotWithUserDTO userTimeSlotWithUserDTO = new UserTimeSlotWithUserDTO();
+        userTimeSlotWithUserDTO.setUser(userDTO);
+        userTimeSlotWithUserDTO.setStartDate(time.getStartDate());
+        switch (userDTO.getInterviewTime()) {
+            case HALF_HOUR: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(30));
+                break;
+            }
+            case HOUR: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(60));
+                break;
+            }
+            case HOUR_HALF: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(90));
+                break;
+            }
+        }
+        userTimeSlotRepository.save(mapper.toTimeForCallUserEntity(userTimeSlotWithUserDTO));
+        log.info("User time was restoring");
+    }
+
+    private void checkTime(UserTimeSlotDTO time) {
+        log.info("Bringing time to a common form");
+        time.setStartDate(commonTimeForm(time.getStartDate()));
+        time.setEndDate(commonTimeForm(time.getEndDate()));
+        log.info("Time brought to a common form");
+    }
+
+    private LocalDateTime commonTimeForm(LocalDateTime dateTime) {
+        if (dateTime.getMinute() % 60 < 30) {
+            return  LocalDateTime.of(dateTime.getYear(), dateTime.getMonth(),
+                    dateTime.getDayOfMonth(), dateTime.getHour(),
+                    DEFAULT_START_MINUTES_IF_BETWEEN_ZERO_THIRTY);
+        }
+        if (dateTime.getMinute() % 60 > 30){
+            return  LocalDateTime.of(dateTime.getYear(), dateTime.getMonth(),
+                    dateTime.getDayOfMonth(), dateTime.getHour() + 1,
+                    DEFAULT_START_MINUTES_IF_BETWEEN_THIRTY_ZERO);
+        }
+        throw new RuntimeException("Time exception can't to do common form for time");
+    }
+
+    private void separateTime(UserTimeSlotDTO time, InterviewTime interviewTimeUser, List<UserTimeSlotWithUserDTO> resultUSerTimeList) {
+        log.info("Separate time on part");
+        List<UserTimeSlotWithUserDTO> newUserTimeList = new ArrayList<>();
+        Duration duration = Duration.between(time.getStartDate(), time.getEndDate());
+        long numberOfPeriods = determineTime(duration, interviewTimeUser);
+        for (int i = 0; i < numberOfPeriods; i++) {
+            UserTimeSlotWithUserDTO tempUserTime = new UserTimeSlotWithUserDTO();
+            tempUserTime.setStartDate(time.getStartDate());
+            tempUserTime.setEndDate(tempUserTime.getStartDate().plusMinutes(INTERVIEW_TIME));
+            newUserTimeList.add(tempUserTime);
+            time.setStartDate(tempUserTime.getEndDate());
```

Comment:
I don't support modification with input data. For me any data which come to Controller is immutable and they are available for copy, not modification. So it'll be better to copy to local variable `time.getStartDate()` and do modification with it.

Reply:
done